### DISCUSSION TITLE
Integration of Vike (without builtin renderer) with Tanstack Query v5

### DISCUSSION THREAD

**Discussion by phonzammi** [vikejs/vike#1253] [Discussion] [answered]
Hi guys, @brillout, @AurelienLourot and anyone else

I've tried again integrating vike with tanstack-query v5, this time without builtin renderer.
I modified an example : react-v1-full.

My repo [vike-react-tanstack-query](https://github.com/phonzammi/vike-react-tanstack-query)

The patterns I used : 

1. Use global `/renderer/onBeforeRender.tsx` for simple or generic query (exported with config.prefetchQuery).
* Prefetching with global `onBeforeRender()` can be apply to any page with a simple or generic query, e.g. prefetching all posts or movies, etc.
* Example : the `posts/index/+prefetchQuery.ts` exports the required query for `posts/index/+Page.tsx`, this query will be prefetched in the global `onBeforeRender()` (`/renderer/+onBeforeRender.ts`). 

2. Use local `onBeforeRender.tsx` inside each pages for complex query.
* Fetching/Prefetching with local `onBeforeRender()` only apply to a page for complex query, e.g. depends on routeParams, filters, etc.
* Example: the `/posts/@id/onBeforeRender.tsx` to fetch/prefetch a single post for `/posts/@id/+Page.tsx` with id param. 

I didn't mean to disturb your time, but
Please take a look and review it, I'm waiting for any suggestions.

FYI @JoshShearer

**Comment by lourot** [comment, marked as answer]
Hey @phonzammi this looks great at the first glance. Thanks a lot. Adding a link to it in the docs: https://github.com/vikejs/vike/pull/1255

**Reply by phonzammi** [reply]
Thanks @AurelienLourot
